javalib/src/main/scala/java/nio/channels/SelectorImpl.scala

```diff
@@ -0,0 +1,158 @@
+package java.nio.channels
+
+import java.io.IOException
+import java.net.Net
+import java.nio.channels.spi.{
+  AbstractSelectableChannel,
+  AbstractSelectionKey,
+  AbstractSelector,
+  SelectorProvider
+}
+import java.util
+import scala.collection.mutable
+import scala.scalanative.libc.stdlib.{malloc, realloc}
+import scala.scalanative.meta.LinktimeInfo.isWindows
+import scala.scalanative.unsafe.{CSize, Ptr, sizeof}
+import scala.scalanative.unsigned._
+import scalanative.posix.poll._
+import scalanative.posix.pollOps._
+
+private[channels] final class SelectorImpl(provider: SelectorProvider)
+    extends AbstractSelector(provider) {
+
+  private val keyMap = mutable.Map.empty[SelectionKey, Int]
+  private val selectedKeySet = mutable.Set.empty[SelectionKey]
+
+  private[channels] val keysLock: Object = new Object
+
+  private var pollSize: CSize = 8.toUInt
+  private var pollArr: Ptr[struct_pollfd] = malloc(
```

Comment:
This may be relevant other places in the 41 files. 

Why a `malloc()` here?  That immediately sends me off
looking for a corresponding `free()`.  In a two minute search
I found a `realloc()`,  but no `free()`. The longer I have to work to
find it, the less confidence I have in both the correctness of the code
and of my code searching skills.

I'd have to study the code in more detail, after I finish the other files
but I believe normal array could be used here (and in doublePollArr())
Then at the call sites the `at(0)` idiom could be used (may need an `unsafe` package include) to get a pointer: `pollArr.at(0)`.

I have checked other places which use `poll` for one fd and
the use of `poll` rather than `epoll` is reasonable there. Here,
I would want to trace to see if `epoll` is a candidate, at least in
theory.

Now the question comes down to Software Engineering and 
contributor's preference.

One way of dealing with this is to do a 'reduction in force'. That is,
create and Issue which describes this reviewer concern and then
let this PR flow through. That means that folks are dealing with
one file instead of 41.   

I suspect there is no `MumbleSelectorTest` yet which exercises
this code and, if it were flakey, cause intermittent Segmentation
Faults in CI.

If that is true, I suggest that alternative.

The other alternative, which some contributors like and some reviewers prefer/insist_upon is "Fix it where and when you find it! Do not let 
cruft into Git. Devos come & go and deferred work may never get done".  If this is on a critical path, I concur.

Devo's choice